File: Trackify_Admin/test_ai_support.js
import { initializeApp } from 'firebase/app';
import { getFirestore, collection, addDoc, serverTimestamp } from 'firebase/firestore';

const firebaseConfig = {
  apiKey: "AIzaSyDC9fZJTOi455NsbN2AQRnGz_cjZvmwedc",
  authDomain: "ys-construction.firebaseapp.com",
  projectId: "ys-construction"
};

const app = initializeApp(firebaseConfig);
const db = getFirestore(app);

async function inject() {
  await addDoc(collection(db, 'support_tickets'), {
    issue: "My app keeps crashing when I scan a QR code! I'm stuck on site, please help me immediately!",
    status: 'Open',
    priority: 'High',
    userId: 'user_123',
    history: [{
      type: 'status_change',
      text: 'Ticket created',
      timestamp: new Date().toISOString()
    }],
    createdAt: serverTimestamp()
  });
  console.log("Injected Support Ticket for AI Auto-Responder!");
  process.exit(0);
}

inject();

File: test_ai_support.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore
import time

try:
    cred = credentials.Certificate('Trackify_Admin/serviceAccountKey.json')
    firebase_admin.initialize_app(cred)
except ValueError:
    pass

db = firestore.client()

db.collection('support_tickets').add({
    'issue': "My app keeps crashing when I scan a QR code! I'm stuck on site, please help me immediately!",
    'status': 'Open',
    'priority': 'High',
    'userId': 'user_123',
    'history': [{
      'type': 'status_change',
      'text': 'Ticket created',
      'timestamp': firestore.SERVER_TIMESTAMP
    }],
    'createdAt': firestore.SERVER_TIMESTAMP
})

print("Injected Support Ticket!")
